### PR TITLE
Feat/wishlist

### PR DESCRIPTION
**Pull request by manish591**
### Added fully functional wishlist feature. 

This PR contains the following features
* Get data from the server if a user is logged in, otherwise redirecting the user to the login page.
* Add to wishlist, user can add items to the wishlist by clicking the wishlist button. While making the request to the server I have disabled the button so that the user can't perform another request between ongoing server calls. 
* Remove item from the wishlist.

https://user-images.githubusercontent.com/79859472/159152897-8c317115-24ce-43cf-a301-66a9eb3f6054.mp4

### CODE REVIEW

File: src/pages/wishlist/WishlistCard.jsx [hunk 1]
```diff
@@ -9,23 +18,36 @@ const WishlistCard = () => {
...
+        <button
+          className="card__remove-wishlist"
+          onClick={() =>
+            removeItemFromWishlist({
+              _id,
+              currentUser,
+              stateDispatch,
+              setIsLoading,
+            })
+          }
```
Comment (romabulani) [inline src/pages/wishlist/WishlistCard.jsx:30]: I think you can use `useContext` in your custom hook useAppActions, so you wont have the need to pass these arguments in the function

**Review by romabulani — APPROVED** [review]
- Liked the custom hooks
- naming is self-explanatory

From your demo video, I can see that user can not remove the item from wishlist on the products page. But its just only my suggestion, that you should give user the option to save and remove from wishlist on products page as well

**Review by samyakshah3008 — APPROVED** [review]
Good naming conventions 
Loved the wishlist functionality 
Clean code and UI ✅ 
Really liked your documentation of PR preview, that made reviewing process easy 

Minor suggestion: You can add some comments in your code for better review experience

File: src/pages/wishlist/WishlistCard.jsx (hunk 1 above)
Comment (manish591) [reply src/pages/wishlist/WishlistCard.jsx:30]: Users can remove items from the wishlist. I didn't show this in the video. 
Will use Context in the hook, thanks for the suggestion.

**Comment by manish591** [comment]
> Good naming conventions Loved the wishlist functionality Clean code and UI ✅ Really liked your documentation of PR preview, that made reviewing process easy
> 
> Minor suggestion: You can add some comments in your code for a better review experience

Thanks, nice point, will add comments

**Pull request merged**